### PR TITLE
[tests - resource packages] Use extension method instead of constructor in Tests

### PR DESCRIPTION
**Pull request by ysolomchenko**
Design discussion issue #
[discussion](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1943#discussion_r1671348344)

## Changes
Instead of instantiating objects via constructors, a static extension method is now used in tests to enhance code readability and maintainability

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by Kielek** [comment]
@cijothomas, FYI

**Comment by cijothomas** [comment]
Thanks! Looks good. Will wait for PR to be marked undraft for marking approval.

**Review by cijothomas — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**
